### PR TITLE
Call iterators' finalizers in Evaluator's finalizer

### PR DESCRIPTION
**Pull request by kuenishi**
Squashed version of https://github.com/chainer/chainer/pull/4137 .

### CODE REVIEW

File: chainer/training/extensions/evaluator.py [hunk 1]
```diff
@@ -193,3 +193,14 @@ def evaluate(self):
             summary.add(observation)
 
         return summary.compute_mean()
+
+    def finalize(self):
+        """Finalizes the evaluator object.
+
+        This method calls the `finalize` method of each iterator that
+        this updater has.
```
Comment (kmaehashi) [inline chainer/training/extensions/evaluator.py]: s/updater/evaluator/

**Comment by kuenishi** [comment]
Fix typo, and next time I'll use my own forked repo to avoid further typoes remain in commit logs.

**Comment by kmaehashi** [comment]
LGTM!

**Comment by kuenishi** [comment]
Thank you~

**Pull request merged**